src/app/modules/search-config/services/search-global-form-builder/search-global-form-builder.service.ts

```diff
@@ -17,43 +17,76 @@
 import { marker } from '@biesbjerg/ngx-translate-extract-marker';
+import { SearchCollectionComponent } from '@search-config/components/search-collection/search-collection.component';
 import { CollectionService } from '@services/collection-service/collection.service';
+import { CollectionField } from '@services/collection-service/models';
 import { toKeywordOptionsObs, toTextOptionsObs } from '@services/collection-service/tools';
 import { DefaultValuesService } from '@services/default-values/default-values.service';
 import { MainFormService } from '@services/main-form/main-form.service';
-import { ConfigFormGroup, InputFormControl, SelectFormControl, SelectOption, SliderFormControl } from '@shared-models/config-form';
+import { CollectionConfigFormGroup } from '@shared-models/collection-config-form';
+import {
+  ComponentFormControl, ConfigFormGroup, InputFormControl,
+  SelectFormControl, SelectOption, SliderFormControl
+} from '@shared-models/config-form';
 import { Observable } from 'rxjs';
 
-export class SearchGlobalFormGroup extends ConfigFormGroup {
-
+export class SearchCollectionFormGroup extends CollectionConfigFormGroup {
   public constructor(
+    collection: string,
+    collectionService: CollectionService,
     textFieldsObs: Observable<Array<SelectOption>>,
     keywordFieldsObs: Observable<Array<SelectOption>>
   ) {
-    super(
+    super(collection,
       {
-        name: new InputFormControl(
-          null,
-          marker('Placeholder'),
-          marker('Placeholder descritpion'),
-          null,
-          { title: 'Search' }),
+        collection: new SelectFormControl(
+          collection,
+          marker('Collection'),
+          undefined,
+          false,
+          collectionService.getCollections().map(c => ({ label: c, value: c })),
+          {
+            optional: false,
+            resetDependantsOnChange: true
+          },
+        ),
         searchField: new SelectFormControl(
           null,
           marker('Search field'),
-          marker('Search field description'),
+          undefined,
           true,
           textFieldsObs
         ),
         autocompleteField: new SelectFormControl(
           null,
           marker('Autocomplete field'),
-          marker('Autocomplete field description'),
+          undefined,
           true,
           keywordFieldsObs
         ),
+      }
+    );
+  }
+  public customControls = {
+    collection: this.get('collection') as SelectFormControl,
+    searchField: this.get('searchField') as SelectFormControl,
+    autocompleteField: this.get('autocompleteField') as SelectFormControl,
+  };
+}
+export class SearchGlobalFormGroup extends ConfigFormGroup {
+
+  public constructor(
+  ) {
+    super(
+      {
+        name: new InputFormControl(
+          null,
+          marker('Placeholder'),
+          marker('Placeholder descritpion'),
+          null,
+          { title: 'Search' }),
```

Comment:
Missing marker + does this work ? I think there  is a TODO above `title` in the extraConfig of forms